src/nerc_rates/models.py

```diff
@@ -32,8 +34,22 @@ def validate_date_range(cls, data: Self):
         return data
 
 
+def validate_rate_type(v: Any) -> str | None:
+    allowed = {"str", "Decimal", "bool"}
+    if v is not None and v not in allowed:
+        raise ValueError(f'type must be one of {allowed}')
+    return v
+
+
+class RateType(StrEnum):
+    STR = "str"
+    DECIMAL = "Decimal"
+    BOOL = "bool"
+
+
 class RateItem(Base):
     name: str
+    type: RateType
```

Comment:
Now that I think about it, could you add a class validation method here similar to `validate_no_overlap` that checks to see that each `RateValue` actually conforms to its type? It would look something like this:
```suggestion python
@pydantic.model_validator(mode="after")
@classmethod
def validate_rate_type(cls, data: Self):
    rate_type = {"str": str, "bool": bool, "Decimal": Decimal}.get(
        data.type
    )
    for x in data.history:
        if rate_type == Decimal:
            try:
                rate_type(x.value)
            catch:
                raise ValueError(f"{x} is not valid Decimal")
        elif rate_type == bool:
            if x.value not in ["True", "False"]:
                raise ValueError(f"Bool field must be a string of either True or False, got {x.value}")

    return data
```

This is to make sure that as people edit `rates.yaml`, this restricts what values they can enter and makes sure its valid. Could you also add a test case for this as well?

Reply:
@QuanMPhm 
elif rate_type == bool:
                if x.value.lower() not in ["true", "false"]:
                    raise ValueError(f"Bool field must be a string of either True or False, got {x.value}")
        return data
        
     This check should be case insensitive I believe. That's how you were originally calling it.